pkg/ccl/streamingccl/streamproducer/event_stream.go

```diff
@@ -324,6 +332,48 @@ func (p *checkpointPacer) shouldCheckpoint(
 	return false
 }
 
+// Add a RangeFeedSSTable into current batch and return size of the bytes added.
+func (s *eventStream) addSST(sst *roachpb.RangeFeedSSTable, batch *streampb.StreamEvent_Batch) (int, error) {
+	for _, sp := range s.spec.Spans {
+		if sp.Contains(sst.Span) {
+			batch.Ssts = append(batch.Ssts, *sst)
+			return sst.Size(), nil
+		}
+	}
+
+	// If span of the sst exceeds boundaries of the watched spans,
+	// we trim the sst data to avoid sending unnecessary data.
+	iter, err := storage.NewMemSSTIterator(sst.Data, true)
+	if err != nil {
+		return 0, err
+	}
+	defer iter.Close()
+	size := 0
+	for _, sp := range s.spec.Spans {
```

Comment:
I don't think we guaranteed Spans sorted.. Would it make sense to sort them using SpangGroup first?